runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -667,20 +668,18 @@ MM_CopyForwardScheme::calculateObjectDetailsForCopy(MM_ScavengerForwardedHeader*
 	}
 
 	/* IF the object has been hashed and has not been moved, then we need generate hash from the old address */
-	*doesObjectNeedHash = (forwardedHeader->hasBeenHashed() && !forwardedHeader->hasBeenMoved());
-	*isObjectGrowingHashSlot = false;
+	UDATA forwardedHeaderPreservedSlot = forwardedHeader->getPreservedSlot();
```

Comment:
This assumes that preserved flags are lowest bits of preserved slot, what actually is true. Still, it's more future proof to extract the flags and then proved them later to hasBeenMoved and other APIs:

forwardedHeaderPreservedFlags = getPreservedFlags(forwardedHeader);
....